ravem/indico_ravem/operations.py

```diff
@@ -153,43 +160,67 @@ def disconnect_room(room_name, vc_room, force=False, room_special_name=None):
+    vc_room_id = service_api.get_room_id(vc_room.data)
     if not status['connected']:
         raise RavemOperationException(
-            _("The room {room} is already disconnected.").format(room=room_special_name),
+            _("The room {room} is already disconnected.").format(room=_room_name),
             'already-disconnected'
         )
-    if status['vc_room_name'] != vc_room.name:
+    if status['vc_room_id'] != vc_room_id:
         if not force:
             raise RavemOperationException(
-                _("The room {room} is connected to an other Vidyo room: {status[vc_room_name]}")
-                .format(room=room_special_name, status=status),
+                _("The room {room} is connected to another videoconference room: {vc_room}")
+                .format(room=_room_name, vc_room=status['vc_room_id']),
                 'connected-other'
             )
         else:
-            RavemPlugin.logger.info("Force disconnect of room %s from vc_room %s "
-                                    "(expected to disconnect from vc_room %s)",
-                                    room_special_name, status['vc_room_name'], vc_room.name)
+            RavemPlugin.logger.info("Force disconnect of room %s from videoconference %s",
+                                    _room_name, status['vc_room_id'])
 
-    response = disconnect_endpoint(room_name, status['vc_room_name'], status['service_type'])
+    response = service_api.disconnect_endpoint(room_name, vc_room_id)
 
-    if 'error' in response:
+    if response.get('error'):
         if response['error'] == 'Call already disconnected':
             raise RavemOperationException(
-                _("The room {room} is already disconnected.").format(room=room_special_name),
+                _('The room {room} is already disconnected.').format(room=_room_name),
                 'already-disconnected'
             )
 
-        RavemPlugin.logger.error("Failed to disconnect the room %s from the Vidyo room %s with error: %s",
-                                 room_special_name, vc_room.name, response['error'])
+        RavemPlugin.logger.error("Failed to disconnect the room %s from the videoconference room %s with error: %s",
+                                 _room_name, vc_room_id, response['error'])
         raise RavemException(
-            _("Failed to disconnect the room {room} from the Vidyo room {vc_room.name} with error: {response[error]}")
-            .format(room=room_special_name, vc_room=vc_room, response=response)
+            _("Failed to disconnect the room {room} from the videoconference room {vc_room} with error: "
+              "{response[error]}").format(room=_room_name, vc_room=vc_room_id, response=response)
+        )
+
+
+def get_api(service_type):
+    if service_type not in API:
+        raise RavemException(
+            _("The videoconference service {service} is not supported").format(
+                service=service_type
+            )
```

Comment:
I don't think this need to be translatable, it will never happen during regular usage